filebeat/tests/system/test_modules.py

```diff
@@ -194,6 +194,9 @@ def _test_expected_events(self, test_file, objects):
                 for k, obj in enumerate(objects):
                     objects[k] = self.flatten_object(obj, {}, "")
                     clean_keys(objects[k])
+                    for key in objects[k].keys():
+                        if isinstance(objects[k][key], list) and len(objects[k][key]) > 0 and not isinstance(objects[k][key][0], dict):
```

Comment:
I'm a little worried about the last check to see if the first element is a dict, if any of the items in the list can't be compared to one another with "<" we will get a runtime error.  What about switching to:
```
if isinstance(objects[k][key], list):
    objects[k][key].sort(key=str)
```

That will force string comparison for everything.

What do you think?

Reply:
I never thought about forcing everything to a string. That's a really interesting idea. I'll try that tomorrow.

Reply:
It seems to be working, but I'm running into an issue that I had before.  When running the test against all the modules `GENERATE=true mage -v pythonIntegTest`, I end up with some errors below.  Any thoughts?
![image](https://user-images.githubusercontent.com/13125104/117009116-051e6900-acb1-11eb-89d2-e5b2cef2049c.png)

Reply:
Things work fine when I do modules individually

Reply:
Weird.  I'm not seeing those errors when I run your PR locally.

anything in the `build/system-tests/run/<failed test name>` directory?  It looks like some documents got left in elasticsearch `expected event.module=kafka but got redis`.

Might need to clean out docker images and re-run the test.

Reply:
Ok, I'll schwack them and start clean

Reply:
still got the same error.  This is what shows in the `output.log` file under `build/system-tests/run/<failed test name>`
```
2021-05-04T21:36:37.226-0200	INFO	instance/beat.go:651	Home path: [/go/src/github.com/elastic/beats/filebeat] Config path: [/go/src/github.com/elastic/beats/filebeat] Data path: [/go/src/github.com/elastic/beats/filebeat/data] Logs path: [/go/src/github.com/elastic/beats/filebeat/logs]
2021-05-04T21:36:37.237-0200	DEBUG	[beat]	instance/beat.go:704	Beat metadata path: /go/src/github.com/elastic/beats/filebeat/data/meta.json
2021-05-04T21:36:37.237-0200	INFO	instance/beat.go:659	Beat ID: 32ac6545-bf91-437e-8aa3-a64e3359883a
2021-05-04T21:36:37.238-0200	INFO	instance/beat.go:407	filebeat stopped.
2021-05-04T21:36:37.238-0200	ERROR	instance/beat.go:963	Exiting: data path already locked by another beat. Please make sure that multiple beats are not sharing the same data path (path.data).
Exiting: data path already locked by another beat. Please make sure that multiple beats are not sharing the same data path (path.data).
/go/src/github.com/elastic/beats/filebeat/filebeat.test -systemTest -e -d * -once -c /go/src/github.com/elastic/beats/filebeat/build/system-tests/run/test_modules.Test.test_fileset_file_076_elasticsearch/filebeat.yml -E setup.ilm.enabled=false -modules=elasticsearch -M elasticsearch.*.enabled=false -M elasticsearch.gc.enabled=true -M elasticsearch.gc.var.input=file -M elasticsearch.gc.var.paths=[/go/src/github.com/elastic/beats/filebeat/module/elasticsearch/gc/test/test.log] -M *.*.input.close_eof=true
```

Reply:
That looks like filebeat didn't exit before the next test of a module started.

I'm going to approve and see if we get the same errors with CI.

Reply:
ok, I still would like to have the generated data files updated so follow on PRs don't have to worry about the unrelated changes to the data files. So if you're able to run all the modules together and update the files, can you push those changes to the branch??